pkg/resources/resource_monitor.go

```diff
@@ -283,13 +277,54 @@ func extractTriggerInts(s sql.NullString) ([]int, error) {
 	return out, nil
 }
 
+// inSlice returns true if n is in string slice h otherwise false.
+func inSlice(n string, h []string) bool {
+	for _, v := range h {
+		if v == n {
+			return true
+		}
+	}
+	return false
+}
+
+// convertTerraformSetToStringSlice turns a terraform set into a string slice.
+func convertTerraformSetToStringSlice(i interface{}) []string {
+	s := make([]string, len(i.(*schema.Set).List()))
+	for x, w := range i.(*schema.Set).List() {
+		s[x] = w.(string)
+	}
+	return s
+}
+
+// compareTerraformSets compares two terraform sets and returns a string slice of all values in the first set that is
+// not in the second set.
+func compareTerraformSets(firstSet interface{}, secondSet interface{}) []string {
+	res := make([]string, 0)
+	sliceOne := convertTerraformSetToStringSlice(firstSet)
+
+	sliceTwo := convertTerraformSetToStringSlice(secondSet)
+
+	for _, s := range sliceOne {
+		if !inSlice(s, sliceTwo) {
+			res = append(res, s)
+		}
+	}
+	return res
+}
+
+// UpdateResourceMonitor implements schema.UpdateFunc.
 func UpdateResourceMonitor(d *schema.ResourceData, meta interface{}) error {
 	db := meta.(*sql.DB)
 	id := d.Id()
 
 	stmt := snowflake.NewResourceMonitorBuilder(id).Alter()
```

Comment:
this was a bit confusing since its not actually the statement (which would be a string), but the alterBuilder instead. This should be renamed for clarity